src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -659,8 +669,8 @@ class Extensions extends Disposable {
 		return this.local;
 	}
 
-	async syncInstalledExtensionsWithGallery(galleryExtensions: IGalleryExtension[], productVersion: IProductVersion): Promise<void> {
-		const extensions = await this.mapInstalledExtensionWithCompatibleGalleryExtension(galleryExtensions, productVersion);
+	async syncInstalledExtensionsWithGallery(galleryExtensions: IGalleryExtension[], productVersion: IProductVersion, flagMissingFromGallery: boolean): Promise<void> {
+		const { extensions, missing } = await this.mapInstalledExtensionWithCompatibleGalleryExtension(galleryExtensions, productVersion);
 		for (const [extension, gallery] of extensions) {
 			// update metadata of the extension if it does not exist
 			if (extension.local && !extension.local.identifier.uuid) {
```

Comment:
You can set the `missingFromGallery` in this for loop already. Also you do not need to have `missing` extensions to find if an extension is missing gallery or not. You can check in following way

```ts
if (flagMissingFromGallery) {
	if (extension.identifier.uuid && !extension.gallery) {
		extension.missingFromGallery = true;
	}
}
```